src/languages/c.js

```diff
@@ -54,15 +54,16 @@ export default function(hljs) {
     ]
   };
 
-  const NUMBERS = {
-    className: 'number',
-    variants: [
-      { begin: '\\b(0b[01\']+)' },
-      { begin: '(-?)\\b([\\d\']+(\\.[\\d\']*)?|\\.[\\d\']+)((ll|LL|l|L)(u|U)?|(u|U)(ll|LL|l|L)?|f|F|b|B)' },
-      { begin: '(-?)(\\b0[xX][a-fA-F0-9\']+|(\\b[\\d\']+(\\.[\\d\']*)?|\\.[\\d\']+)([eE][-+]?[\\d\']+)?)' }
```

Comment:
Where did `[eE]` go in the new code?

Reply:
Are `p` and `e` exclusive? If so could we split this into two variants?

Reply:
we could but the new commit i made checks for both using a regex or

Reply:
Sure, but is then becomes longer and harder to read. Let's have two different ones here, please.

Reply:
i've split the  regex into two  @joshgoebel please review changes